packages/transaction_signing_gateway/lib/encrypt/aes_cipher.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:encrypt/encrypt.dart';
+import 'package:transaction_signing_gateway/encrypt/cipher.dart';
+
+class AESCipher implements Cipher {
+  @override
+  String decrypt({required String password, required String encryptedData}) {
+    final encrypter = _fromPassword(password);
+    final bundle = _EncryptedBundle.fromString(encryptedData);
+    if (bundle == null) {
+      throw ArgumentError("encrypted data has wrong format");
+    }
+    return encrypter.decrypt(
+      Encrypted.fromBase64(bundle.encryptedData),
+      iv: IV.fromBase64(bundle.iv),
+    );
+  }
+
+  @override
+  String encrypt({required String password, required String data}) {
```

Comment:
updated the code, now when encrypting we're generating random, 16bytes long IV, use it to encrypt the data and prepend it to the encrypted string.

Reply:
Again I'm not familiar with Dart language, hence not too comfortable reading the code. It seems the IV is generated from the password, we should generate the IV as a random nonce of 128 bits. And use a new unique random IV for each encryption. Salt is 256 bits long? Can't figure that out. 
Salt and IV should not be in Base64 format when used in encryption/derivation techniques, it decreases the entropy. 
See my message in Slack #dev-mobile where I describe the encryption/decryption flows (pseudo code) with the recommendations. Thanks!

Reply:
currently, IV is generated for each encryption using SecureRandom, 128bit long (16 bytes long), password is not taken into consideration when generating it.
```dart
IV _iv() => IV.fromSecureRandom(16);
```
Salt is also generated from SecureRandom as 32bytes long array:
```dart
IV _salt() => IV.fromSecureRandom(32);
```

I've updated the code to do base64 encryption only as the last step before spitting out the encryption result. We base64 encode IV and salt to be able to represent them as strings. they all get concatenated together into the final result of encryption so that we can decrypt the message afterwards. i.e:

```dart
"${base64(IV)} | ${base64(Salt)} | ${base64(EncryptedMessage)}"
````